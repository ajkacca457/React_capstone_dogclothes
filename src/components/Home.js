import React from 'react';
import { Link } from 'react-router-dom';
import Tvshow1 from '../images/tvshow1.jpeg';
import Tvshow2 from '../images/tvshow2.jpg';

const Home = () => (
  <div className="home">
    <div className="content d-block d-sm-flex justify-content-between align-items-center">
      <div className="leftcont text-white text-center">
        <h2>TVWORLD</h2>
        <h3 className="my-2">Home of Your Entertainment</h3>
        <Link to="/shows" className="btn btn-warning rounded">Browse Shows</Link>
      </div>

      <div className="vr" />

      <div className="rightcont">
        <img src={Tvshow1} alt="Show1" className="tvshow s1" />
        <img src={Tvshow2} alt="Show1" className="tvshow s2" />
      </div>

    </div>

  </div>
);

export default Home;
