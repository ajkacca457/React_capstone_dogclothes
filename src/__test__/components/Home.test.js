import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../../components/Home';

describe('Home Component', () => {
  it('renders correctly', () => {
    const hometree = TestRenderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      )
      .toJSON();
    expect(hometree).toMatchSnapshot();
  });

  test('renders the title text of the Home page', () => {
    const { getByText } = render(<BrowserRouter><Home /></BrowserRouter>);
    const title = getByText('TVWORLD');
    expect(title).toBeInTheDocument();
  });

  test('renders the subheading in the page', () => {
    const { getByText } = render(<BrowserRouter><Home /></BrowserRouter>);
    const title = getByText('Home of Your Entertainment');
    expect(title).toBeInTheDocument();
  });
});
