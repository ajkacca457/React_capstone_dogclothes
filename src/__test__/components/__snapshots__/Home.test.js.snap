// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div
  className="home"
>
  <div
    className="content d-block d-lg-flex justify-content-between align-items-center"
  >
    <div
      className="leftcont text-white text-center"
    >
      <h2>
        TVWORLD
      </h2>
      <h3
        className="my-2"
      >
        Home of Your Entertainment
      </h3>
      <a
        className="btn btn-warning rounded"
        href="/shows"
        onClick={[Function]}
      >
        Browse Shows
      </a>
    </div>
    <div
      className="vr d-none d-lg-block"
    />
    <div
      className="rightcont d-none d-lg-block"
    >
      <img
        alt="Show1"
        className="tvshow s1"
        src="tvshow1.jpeg"
      />
      <img
        alt="Show1"
        className="tvshow s2"
        src="tvshow2.jpg"
      />
    </div>
  </div>
</div>
`;
